test/lib/NonLinearDutchDecayLib.t.sol

```diff
@@ -2,133 +2,193 @@
 pragma solidity ^0.8.0;
 
 import {Test} from "forge-std/Test.sol";
-import {NonLinearDutchDecayLib} from "../../src/lib/NonLinearDutchDecayLib.sol";
-import {NonLinearDutchOutput, NonLinearDutchInput, NonLinearDecay} from "../../src/lib/NonLinearDutchOrderLib.sol";
-import {Util} from "../../src/lib/Util.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {console} from "forge-std/console.sol";
+import {NonlinearDutchDecayLib, InvalidDecayCurve} from "../../src/lib/NonlinearDutchDecayLib.sol";
+import {V3DutchOutput, V3DutchInput, NonlinearDutchDecay} from "../../src/lib/V3DutchOrderLib.sol";
+import {Uint16Array, toUint256} from "../../src/types/Uint16Array.sol";
 import {ArrayBuilder} from "../util/ArrayBuilder.sol";
+import {CurveBuilder} from "../util/CurveBuilder.sol";
+import {NegativeUint} from "../../src/lib/MathExt.sol";
 
-/// @notice mock contract to test NonLinearDutchDecayLib functionality
-contract MockNonLinearDutchDecayLibContract {
-    function decay(NonLinearDecay memory curve, uint256 startAmount, uint256 decayStartBlock) public view {
-        NonLinearDutchDecayLib.decay(curve, startAmount, decayStartBlock);
+/// @notice mock contract to test NonlinearDutchDecayLib functionality
+contract MockNonlinearDutchDecayLibContract {
+    function decay(NonlinearDutchDecay memory curve, uint256 startAmount, uint256 decayStartBlock) public view {
+        NonlinearDutchDecayLib.decay(curve, startAmount, decayStartBlock);
     }
 }
 
-contract NonLinearDutchDecayLibTest is Test {
-    MockNonLinearDutchDecayLibContract mockNonLinearDutchDecayLibContract = new MockNonLinearDutchDecayLibContract();
+contract NonlinearDutchDecayLibTest is Test, GasSnapshot {
+    MockNonlinearDutchDecayLibContract mockNonlinearDutchDecayLibContract = new MockNonlinearDutchDecayLibContract();
+
+    function testLocateCurvePositionSingle() public {
+        NonlinearDutchDecay memory curve = CurveBuilder.singlePointCurve(1, 0);
+
+        snapStart("V3-LocateCurvePositionSingle");
+        (uint16 prev, uint16 next) = NonlinearDutchDecayLib.locateCurvePosition(curve, 1);
+        assertEq(prev, 0);
+        assertEq(next, 0);
+
+        (prev, next) = NonlinearDutchDecayLib.locateCurvePosition(curve, 2);
+        assertEq(prev, 0);
+        assertEq(next, 0);
+        snapEnd();
```

Comment:
Yeah, since each call can have different gas, I've been measuring the per-call gas by taking the avg over a series of calls (e.g. `.forge-snapshots/V3-ExtendedMultiPointDutchDecay.snap` / 21).